drivers/clock_control/clock_stm32_ll_h7.c

```diff
@@ -321,6 +322,66 @@ static uint32_t get_vco_output_range(uint32_t vco_input_range)
 
 #endif /* ! CONFIG_CPU_CORTEX_M4 */
 
+/** @Brief Verifies clock is part of actve clock configuration */
+static inline int enabled_clock(uint32_t src_clk)
+{
+	int r = 0;
+
+	switch (src_clk) {
+	case STM32_SRC_SYSCLK:
+		break;
+	case STM32_SRC_HSE:
+		if (!IS_ENABLED(STM32_HSE_ENABLED)) {
+			r = -ENOTSUP;
+		}
+		break;
+	case STM32_SRC_LSE:
+		if (!IS_ENABLED(STM32_LSE_ENABLED)) {
+			r = -ENOTSUP;
+		}
+		break;
```

Comment:
suggestion here, maybe this would be more compact

```c
	if ((src_clk == STM32_SRC_SYSCLK) ||
	    ((src_clk == STM32_SRC_HSE) && IS_ENABLED(STM32_HSE_ENABLED)) ||
	    ((src_clk == STM32_SRC_LSE) && IS_ENABLED(STM32_LSE_ENABLED)) ||
	    ...
	    ((src_clk == STM32_SRC_PLL3_R) && IS_ENABLED(STM32_PLL3_R_ENABLED))){
		return 0;
	}

	return -ENOTSUP;
```

Reply:
Indeed. Done. I've converted U5 as well. 
On `_common.c` there are intermixed `#ifdef FOO` that don't show up well in the `if` format. Maybe that's a matter of taste, but anyway I've kept as is.